openmc/cmfd.py

```diff
@@ -0,0 +1,3072 @@
+        reflector, the map is:
+
+        ::
+
+            [0, 0, 0, 0,
+             0, 1, 1, 0,
+             0, 1, 1, 0,
+             0, 0, 0, 0]
+
+        Therefore a 2x2 system of equations is solved rather than a 4x4. This
+        is extremely important to use in reflectors as neutrons will not
+        contribute to any tallies far away from fission source neutron regions.
+        A ``1`` must be used to identify any fission source region.
+
+    """
+
+    def __init__(self):
+        self._lower_left = None
+        self._upper_right = None
+        self._dimension = None
+        self._width = None
+        self._energy = None
+        self._albedo = None
+        self._map = None
+
+    @property
+    def lower_left(self):
+        return self._lower_left
+
+    @property
+    def upper_right(self):
+        return self._upper_right
+
+    @property
+    def dimension(self):
+        return self._dimension
+
+    @property
+    def width(self):
+        return self._width
+
+    @property
+    def energy(self):
+        return self._energy
+
+    @property
+    def albedo(self):
+        return self._albedo
+
+    @property
+    def map(self):
+        return self._map
+
+    @lower_left.setter
+    def lower_left(self, lower_left):
+        check_type('CMFD mesh lower_left', lower_left, Iterable, Real)
+        check_length('CMFD mesh lower_left', lower_left, 2, 3)
+        self._lower_left = lower_left
+
+    @upper_right.setter
+    def upper_right(self, upper_right):
+        check_type('CMFD mesh upper_right', upper_right, Iterable, Real)
+        check_length('CMFD mesh upper_right', upper_right, 2, 3)
+        self._upper_right = upper_right
+
+    @dimension.setter
+    def dimension(self, dimension):
+        check_type('CMFD mesh dimension', dimension, Iterable, Integral)
+        check_length('CMFD mesh dimension', dimension, 2, 3)
+        for d in dimension:
+            check_greater_than('CMFD mesh dimension', d, 0)
+        self._dimension = dimension
+
+    @width.setter
+    def width(self, width):
+        check_type('CMFD mesh width', width, Iterable, Real)
+        check_length('CMFD mesh width', width, 2, 3)
+        for w in width:
+            check_greater_than('CMFD mesh width', w, 0)
+        self._width = width
+
+    @energy.setter
+    def energy(self, energy):
+        check_type('CMFD mesh energy', energy, Iterable, Real)
+        for e in energy:
+            check_greater_than('CMFD mesh energy', e, 0, True)
+        self._energy = energy
+
+    @albedo.setter
+    def albedo(self, albedo):
+        check_type('CMFD mesh albedo', albedo, Iterable, Real)
+        check_length('CMFD mesh albedo', albedo, 6)
+        for a in albedo:
+            check_greater_than('CMFD mesh albedo', a, 0, True)
+            check_less_than('CMFD mesh albedo', a, 1, True)
+        self._albedo = albedo
+
+    @map.setter
+    def map(self, meshmap):
```

Comment:
Nit: `mesh_map` as a parameter name rather than `meshmap` would be slightly more pythonic.